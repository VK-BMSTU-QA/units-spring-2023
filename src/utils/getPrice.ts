import type { PriceSymbol } from '../types';

export const getPrice = (value: number, symbol: PriceSymbol = 'â‚½'): string =>
    `${value.toLocaleString('en-En')} ${symbol}`;
