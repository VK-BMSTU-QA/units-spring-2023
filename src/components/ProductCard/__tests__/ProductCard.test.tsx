import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { ProductCard } from '../ProductCard';
import { getPrice } from '../../../utils';

jest.mock('../../../utils', () => {
    return {
        __esModule: true,
        getPrice: jest.fn(() => '10 ₽'),
    };
});

describe('test product card', () => {
    afterEach(jest.clearAllMocks);

    it('should render correctly', () => {
        const renderCard = render(
            <ProductCard
                category="Для дома"
                description="Настоящая китайская швабра"
                id={1}
                name="Швабра"
                price={100}
                imgUrl="/img.jpg"
            />
        );
        expect(renderCard.asFragment()).toMatchSnapshot();
    });

    it('should call getPrice once', () => {
        expect(getPrice).toHaveBeenCalledTimes(0);
        render(
            <ProductCard
                category="Для дома"
                description="Настоящая китайская швабра"
                id={1}
                name="Швабра"
                price={100}
                imgUrl="/img.jpg"
            />
        );
        expect(getPrice).toHaveBeenCalledTimes(1);
    });

    it('should conatain an image', () => {
        const renderCard = render(
            <ProductCard
                category="Для дома"
                description="Настоящая китайская швабра"
                id={1}
                name="Швабра"
                price={100}
                imgUrl="/img.jpg"
            />
        );
        expect(renderCard.getByRole('img')).toBeInTheDocument();
    });

    it('should not conatain an image', () => {
        const renderCard = render(
            <ProductCard
                category="Для дома"
                description="Настоящая китайская швабра"
                id={1}
                name="Швабра"
                price={100}
            />
        );
        expect(renderCard.queryByRole('img')).not.toBeInTheDocument();
    });
});
