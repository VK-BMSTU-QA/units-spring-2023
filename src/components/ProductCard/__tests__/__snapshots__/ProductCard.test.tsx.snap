// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test product card should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        Швабра
      </h2>
      <div
        class="product-card__description"
      >
        Настоящая китайская швабра
      </div>
      <h1
        class="product-card__price"
      >
        10 ₽
      </h1>
      <div
        class="product-card__category"
      >
        Для дома
      </div>
    </div>
    <img
      alt="Швабра"
      class="product-card__image"
      src="/img.jpg"
    />
  </div>
</DocumentFragment>
`;
